### PR TITLE
Restructure explainers

### PR DESCRIPTION
**Pull request by aheejin**
- Revives the legacy Phase 3 explainer and put it in `proposals/exception-handling/legacy` directory. Given that we still maintain the legacy formal spec in `document/legacy`, I think it makes sense to have the legacy expliner too.                                 
- Moves two auxiliary docs for the legacy proposal (`Exception-formal-overview.md` and `Exception-formal-examples.md`), currently still in `proposals/exception-handling`, to the new `proposals/exception-handling/legacy` directory too.
- Renames `proposals/exception-handling/old` to `proposals/exception-handling/pre-legacy`, signifying this precedes the "legacy" proposal.
- Fixes the CG meeting date from the current explainer.
- Adds more historical context to the beginning of each explainer in `pre-legacy` and `legacy` directories.

### CODE REVIEW

File: proposals/exception-handling/legacy/Exceptions.md [hunk 1]
```diff
@@ -0,0 +1,697 @@
+# Exception handling
+
+This explainer reflects the up-to-date version of the exception handling
```
Comment (rossberg) [inline proposals/exception-handling/legacy/Exceptions.md]:
```suggestion
This explainer reflects the earlier version of the exception handling
```

**Review by rossberg — COMMENTED** [review]
Good idea. I'm a bit undecided whether we should remove the older documents, since they could likewise be interesting to later observers. Perhaps their status can be made clearer by renaming the `old` directory to something more precise, say, `pre-legacy`, and add a clarifying intro sentence to the docs?

**Comment by dschuff** [comment]
+1 I like this idea also. I think we should also put the JS spec (and especially the tests that cover it) somewhere into the repo as well.

**Comment by aheejin** [comment]
@rossberg 
> Good idea. I'm a bit undecided whether we should remove the older documents, since they could likewise be interesting to later observers. Perhaps their status can be made clearer by renaming the `old` directory to something more precise, say, `pre-legacy`, and add a clarifying intro sentence to the docs?

Restored `old/` proposals and renamed it to `pre-legacy/` as you suggested, and added some text on top of each doc to add some historical context. Also added more historical context in the beginning of `legacy/Exceptions.md` too.

**Comment by aheejin** [comment]
@dschuff 
> +1 I like this idea also. I think we should also put the JS spec (and especially the tests that cover it) somewhere into the repo as well.

Yeah we lost the JS API doc too.. I think we need to restructure the directory to restore that. Will do it as a follow-up.

**Review by rossberg — APPROVED** [review]

**Comment by titzer** [comment]
post-merge lgtm

**Pull request merged**